src/main/java/org/logstash/plugins/inputs/http/promtail/PromtailHandler.java

```diff
@@ -0,0 +1,129 @@
+package org.logstash.plugins.inputs.http.promtail;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.protobuf.Timestamp;
+import org.xerial.snappy.Snappy;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import static java.time.format.DateTimeFormatter.ISO_INSTANT;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class PromtailHandler {
```

Comment:
this file is missing comments. please use this one: 
https://github.com/co-brick/logstash-promtail-http-input/blob/main/src/main/java/org/logstash/plugins/inputs/http/promtail/PromtailHandler.java